shopping-cart/shopping-cart-java/README.md

```diff
@@ -1,46 +1,60 @@
 
-## Setup
+## Requirements
 
-To run this application locally you will need access to a Postgres database. We suggest you run it on a docker container but a local or remote native instance will also work.
+This sample requires Kafka and Postgres as external services. They are pre-configured in `docker-compose.yml` file. Meaning that before running the service, you first need to start the services using the following command:
 
-We provide a `docker-compose.yml` file that you can use to run a Postgres database already configured for this application. The docker container will be exposed on port `5432`.
-
-To create the image and start the container, run the command below at the root of this project.
-
-```bash 
+```bash
 docker-compose up -d
 ```
 
-If you prefer to run Postgres natively on your machine, you need to create the database, the user and password yourself. The application expects it to be running on `localhost` on the default port (`5432`), and it expects there to be a database called `shopping_cart`, with a user called `shopping_cart` with password `shopping_cart` that has full access to it. This can be created using the following SQL:
+Postgres will be available on port `5432` and Kafka on port `9092`.
+
+Of course, local or remote instances for both services will also work. See more details below.
+
+### Postgres configuration
+
+This configuration is only necessary if you are not using the instance provided by `docker-compose.yml`.
+
+First, you need to create the database, the user, and the password yourself. The application expects it to be running on `localhost` on the default port (`5432`), and it expects there to be a database called `shopping_cart`, with a user called `shopping_cart` with password `shopping_cart` that has full access to it. This can be created using the following SQL:
 
 ```sql
 CREATE DATABASE shopping_cart;
 CREATE USER shopping_cart WITH PASSWORD 'shopping_cart';
 GRANT ALL PRIVILEGES ON DATABASE shopping_cart TO shopping_cart;
 ```
 
-Once Postgres is setup, you can start the system by running:
+### Kafka Server
+
+This configuration is only necessary if you are not using the instance provided by `docker-compose.yml`.
+
+You need to configure the application to connect to an external Kafka server. Follow [Lagom documentation to see how to do that](https://www.lagomframework.com/documentation/latest/java/KafkaServer.html#Connecting-to-an-external-Kafka-server).
+
+> **Note**: Some of those configurations are already there, so you only need to update them to your correct values.
+
+## Running in dev mode
+
+After setting up all the requirements, to run the application in dev mode, execute the following command:
 
 ```bash
-sbt runAll
+mvn lagom:runAll
 ```
 
-or, if you prefer to use Maven:
+Or, if you are using sbt:
 
 ```bash
-mvn lagom:runAll
+sbt runAll
```

Comment:
I like how you reordered the Maven and sbt commands alphabetically putting Maven first.

Reply:
This is not my dotfiles, though. 😅 

I move it above because this is the Java version. :-)